var/spack/repos/builtin/packages/babelstream/package.py

```diff
@@ -235,183 +464,408 @@ def cmake_args(self):
+        unsupported = any(
+            unsupported_value in options
+            for options in unsupported_options.get(compiler_name, [])
+            if options == unsupported_value
+        )
+        if unsupported:
+            raise InstallError(
+                f"{unsupported_value} is not supported by the {compiler_name} compiler"
+            )
+        # ===================================
+        #               ARM
+        # ===================================
+        if spec.compiler.name == "arm":
+            flags = "-std=f2018 " + pkg.compiler.opt_flags[4] + " -Wall -Wno-unused-variable"
+            flags += "-march=" + str(spec.target)
+
+            config["DOCONCURRENT_FLAG"] = pkg.compiler.openmp_flag  # libomp.so required
+            config["ARRAY_FLAG"] = pkg.compiler.openmp_flag  # libomp.so required
+            config["OPENMP_FLAG"] = pkg.compiler.openmp_flag  # libomp.so required
+            config["OPENACC_FLAG"] = "-fopenacc"
+
+        # ===================================
+        #               AMD
+        # ===================================
+        if spec.compiler.name == "aocc":
+            flags = "-std=f2018 " + pkg.compiler.opt_flags[3] + " -Wall -Wno-unused-variable"
+
+            config["DOCONCURRENT_FLAG"] = pkg.compiler.openmp_flag  # libomp.so required
+            config["ARRAY_FLAG"] = pkg.compiler.openmp_flag  # libomp.so required
+            config["OPENMP_FLAG"] = pkg.compiler.openmp_flag  # libomp.so required
+            config["OPENACC_FLAG"] = "-fopenacc"
+
+        # ===================================
+        #               CRAY
+        # ===================================
+        if spec.compiler.name == "cce":
+            flags = "-e F -O3"
+
+            config["DOCONCURRENT_FLAG"] = "-h thread_do_concurrent -DCRAY_THREAD_DOCONCURRENT"
+            config["ARRAY_FLAG"] = "-h autothread"
+            config[
+                "OPENMP_FLAG"
+            ] = pkg.compiler.openmp_flag  # if clang based it will be -fopenmp else -h omp
+            config["OPENACC_FLAG"] = "-h acc"  # for cpu only -h omp
+
+        # ===================================
+        #               GCC
+        # ===================================
+        if spec.compiler.name == "gcc":
+            flags = "-std=f2018 -O3 "
+            flags += "-Wall -Wno-unused-dummy-argument -Wno-unused-variable "
+            spec_target = "znver3" if str(spec.target) == "zen3" else str(spec.target)
+            flags += "-march=" + spec_target
+
+            config["FCFLAGS"] = flags
+            config["DOCONCURRENT_FLAG"] = "-ftree-parallelize-loops=4"
+            config["OPENMP_FLAG"] = pkg.compiler.openmp_flag
+            config["OPENACC_FLAG"] = "-fopenacc"
+
+        # ===================================
+        #               NVHPC
+        # ===================================
+        if spec.compiler.name == "nvhpc":
+            flags = pkg.compiler.opt_flags[4]  # for -O3
```

Comment:
It looks like this variable is set for every compiler, and only used for the `nvhpc` compiler.

Reply:
Thanks yes will be updating as I said in previous comment, thanks again!